### PR TITLE
Fix ORDER BY expressions which evaluate to the same canonical

### PR DESCRIPTION
**Pull request by kokosing**
Fix ORDER BY expressions which evaluate to the same canonical

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes #8194

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java [hunk 1]
```diff
@@ -114,10 +114,17 @@ public TopNNode map(TopNNode node, PlanNode source, PlanNodeId newNodeId)
     {
         ImmutableList.Builder<Symbol> symbols = ImmutableList.builder();
         ImmutableMap.Builder<Symbol, SortOrder> orderings = ImmutableMap.builder();
+        Set<Symbol> seenCanonicals = new HashSet<>(node.getOrderBy().size());
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java]: Maybe just use `HashMap` instead `ImmutableMap.Builder` for `orderings` and use `Map.hasKey` to check if entry is already present.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java [hunk 2]
```diff
@@ -114,10 +114,17 @@ public TopNNode map(TopNNode node, PlanNode source, PlanNodeId newNodeId)
...
         for (Symbol symbol : node.getOrderBy()) {
             Symbol canonical = map(symbol);
-            symbols.add(canonical);
-            orderings.put(canonical, node.getOrderings().get(symbol));
+            if (seenCanonicals.contains(canonical)) {
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java]: you can just do:
```
if (seenCanonicals.add(canonical)) {
 ...
}
```

**Comment by kokosing** [comment]
Comments applied

**Comment by kokosing** [comment]
Thanks! That was quick.

**Pull request merged**